src/core/net/nd6.hpp

```diff
@@ -525,6 +564,53 @@ class RaFlagsExtOption : public Option, private Clearable<RaFlagsExtOption>
 
 static_assert(sizeof(RaFlagsExtOption) == 8, "invalid RaFlagsExtOption structure");
 
+
+class ND6TxMessage {
+
+    public:
+        /**
+         * Gets the prepared ND6 message as an `Icmp6Packet`.
+         *
+         * @param[out] aPacket   A reference to an `Icmp6Packet`.
+         *
+         */
+        void GetAsPacket(Icmp6Packet &aPacket) const { aPacket.Init(mArray.AsCArray(), mArray.GetLength()); }
+
+        /**
+         * Appends bytes from a given buffer to the ND6 message.
+         *
+         * @param[in] aBytes     A pointer to the buffer containing the bytes to append.
+         * @param[in] aLength    The buffer length.
+         *
+         * @retval kErrorNone    Bytes are appended successfully.
+         * @retval kErrorNoBufs  Insufficient available buffers to grow the message.
+         *
+         */
+        Error AppendBytes(const uint8_t *aBytes, uint16_t aLength);
+
+        /**
+         * Appends a Source/Target Link-Layer Address Option to the ND6 message.
+         *
+         * @param[in] aInfraIfLinkLayerAddress      The InfraIf Link-Layer Address.
+         * @param[in] isSource                      True for Source Link-Layer Address Option
+         *                                          False for Target Link-Layer Address Option.
+         *
+         * @retval kErrorNone    Option is appended successfully.
+         * @retval kErrorNoBufs  Insufficient available buffers to grow the message.
+         *
+         */
+        Error AppendLinkLayerOption(otInfraIfLinkLayerAddress *aInfraIfLinkLayerAddress, bool isSource);
```

Comment:
Use `const &` (reference) to link-layer address insetad of pointer?
```suggestion
        Error AppendLinkLayerOption(const InfraIf::LinkLayerAddress &aInfraIfLinkLayerAddress, bool isSource);
```